tests/cpp/test_learner.cc

```diff
@@ -190,6 +191,77 @@ TEST(Learner, IO) {
   delete pp_dmat;
 }
 
+TEST(Learner, Json_IO) {
+  size_t constexpr kRows = 10;
+  int32_t constexpr kIters = 4;
+  std::vector<bst_float> labels(kRows);
+  for (size_t i = 0; i < labels.size(); ++i) {
+    labels[i] = i;
+  }
+  auto pp_dmat = CreateDMatrix(kRows, 10, 0);
```

Comment:
[Optional] [Minor] Should we use a unique_ptr here?

Reply:
@trams The generator uses c_api to do the job.  It's an headache but so far I don't have the time to refactor our unittests ...